simple/stats/stat_var_hierarchy_generator.py

```diff
@@ -177,21 +193,54 @@ class StatVarHierarchy:
   svg_triples: list[Triple]
 
 
-def _attach_verticals(pop_type_svgs: list[SVG]):
-  # For now, we put all pop type svgs under the root svg.
-  # TODO: pass vertical specs to this function and implement vertical tagging here.
-  for svg in pop_type_svgs:
-    svg.parent_svg_ids[sc.DEFAULT_CUSTOM_ROOT_SVG_ID] = True
-
-
-def _create_hierarchy_root_svg() -> SVG:
-  svg = SVG(sc.DEFAULT_CUSTOM_ROOT_SVG_ID, sc.DEFAULT_CUSTOM_ROOT_SVG_NAME)
-  svg.parent_svg_ids[sc.ROOT_SVG_ID] = True
-  return svg
-
-
-def _get_svgs_with_no_parent(svgs: dict[str, SVG]) -> list[SVG]:
-  return list(filter(lambda svg: not svg.parent_svg_ids, svgs.values()))
+# Attaches matching pop type svgs to vertical svgs, creates those vertical svgs and returns them.
+def _attach_verticals(poptype2svg: dict[str, SVG],
+                      vertical_specs: list[VerticalSpec]) -> dict[str, SVG]:
+  vertical_svgs: dict[str, SVG] = {}
+  for vertical_spec in vertical_specs:
+    pop_type_svg = poptype2svg.get(vertical_spec.population_type)
+    # If no matching pop type svg, skip.
+    if not pop_type_svg:
+      continue
+    # If no vertical spec mprop in svg mprops, skip.
+    if not any(mprop in vertical_spec.measured_properties
+               for mprop in pop_type_svg.measured_properties):
```

Comment:
can `if not (x.measured_properties.keys() & y.measured_properties.keys()):` work?

Reply:
Nice simplification - it sure does. Done.